modeling/06_Landsat_8_GTB.Rmd

```diff
@@ -0,0 +1,550 @@
+ ---
+title: "eePlumB Develop and Apply GTB for Landsat 8"
+author: "ROSSyndicate"
+date: "2024-01-25"
+output: html_document
+editor_options:
+  markdown:
+    wrap: 80
+---
+
+```{r setup, echo = F}
+libs = c('reticulate', 'tidyverse')
+
+package_loader <- function(x) {
+    if (x %in% installed.packages()) {
+      library(x, character.only = TRUE)
+    } else {
+      install.packages(x)
+      library(x, character.only = TRUE)
+    }
+}
+
+lapply(libs, package_loader)
+```
+
+# Purpose
+
+This script develops and applies Gradient Tree Boost Models to the Landsat 8
+image stack.
+
+## Activate conda environment
+
+Check for virtual environment and activate, otherwise, set up virtual
+environment.
+
+```{r, conda env}
+if (!dir.exists("env")) {
+  source("pySetup.R")
+} else {
+  use_condaenv(file.path(getwd(), "env"))
+}
+```
+
+### Settings/modules
+
+Import the needed modules and set model version date
+
+```{python}
+import ee
+import os
+import time
+import pandas as pd
+
+v_date = '2024-01-08'
+```
+
+## GEE Setup
+
+```{python}
+ee.Authenticate()
+```
+
+When your browser states 'Google Earth Engine authentication successful!', the
+authentication is complete. This authentication is valid for 7 days.
+
+Now, we need to initialize our GEE session. You may need to change the project 
+name to one you own if you do not have write access.
+
+```{python}
+ee.Initialize(project = 'ee-ross-superior')
+```
+
+# Import assets
+
+These assets were created in the 03_Train_Test_Split.Rmd file
+
+```{python}
+training_ls8 = ee.FeatureCollection("projects/ee-ross-superior/assets/train-test/training_ls8")
+testing_ls8 = ee.FeatureCollection("projects/ee-ross-superior/assets/train-test/testing_ls8")
+```
+
+## Train the GTB model
+
+```{python}
+ls_input_feat = ["Red", "Green", "Blue", "B5", "B6", "B7"]
+output_label = "class"
+class_values = (['cloud',
+  'openWater',
+  'lightNearShoreSediment',
+  'offShoreSediment',
+  'darkNearShoreSediment'])
+```
+
+### Landsat 8
+
+```{python}
+trainedGTB_ls8 = (ee.Classifier.smileGradientTreeBoost(10).train(
+  features = training_ls8,
+  classProperty = 'byte_property',
+  inputProperties = ls_input_feat
+))
+```
+
+Unfortunately, there is no current mechanism to save the GTB object, so all models
+may be *slightly* different if re-created, but will likely have very similar
+outcomes. 
+
+## Evaluate the models
+
+### Landsat 8
+
+```{python}
+trainingMatrixGTB_ls8 = (trainedGTB_ls8
+  .confusionMatrix())
+
+#convert to pandas dataframe with class info
+training_conf_l7 = (pd.DataFrame(
```

Comment:
I can't suggest edits because it is already merged but I am flagging a few spots that look like we copied and didn't save the correct name.

This one should probably be `training_conf_l8`. I would expect the code to fail later when it calls `training_conf_l8` and it wasn't created so maybe you have already fixed after this was merged.